common/http/platform/beast/http.cpp

```diff
@@ -270,11 +272,19 @@ error::Error Client::Initialize() {
 				return error::Error(
 					ec.default_error_condition(), "Could not load client certificate");
 			}
-			ssl_ctx_[i].use_private_key_file(
-				client_config_.client_cert_key_path, boost::asio::ssl::context_base::pem, ec);
-			if (ec) {
-				return error::Error(
-					ec.default_error_condition(), "Could not load client certificate private key");
+			auto exp_key = crypto::PrivateKey::Load(
+				{client_config_.client_cert_key_path, "", client_config_.ssl_engine});
+			if (!exp_key) {
+				return exp_key.error().WithContext(
+					"Error loading private key from " + client_config_.client_cert_key_path);
+			}
+
+			const int ret =
+				SSL_CTX_use_PrivateKey(ssl_ctx_[i].native_handle(), exp_key.value().Get());
+			if (ret != 1) {
+				log::Error(
+					"Failed to add the PrivateKey: " + client_config_.client_cert_key_path
+					+ " to the SSL CTX. The HTTP client will not be functioning normally");
```

Comment:
I think we should return error here as well, we have opted for returns on all other error paths.